igvc_navigation/src/field_d_planner/FieldDPlanner.h

```diff
@@ -0,0 +1,254 @@
+  void initializeSearch();
+  /**
+  Updates a node's standing in the graph search problem. Update dependant upon
+  the node's g value and rhs value relative to each other.
+
+  Locally inconsistent nodes (g != rhs) are inserted into the priority queue
+  while locally consistent nodes are not.
+
+  @param[in] s Node to update
+  */
+  void updateNode(const Node& s);
+  /**
+  Expands nodes in the priority queue until optimal path to goal node has been
+  found. The first search is equivalent to an A* heuristic search. All calls
+  to computeShortestPath() thereafter only expand the nodes necessary to
+  compute the optimal path to the goal node.
+
+  @return number of nodes expanded in graph search
+  */
+  int computeShortestPath();
+  /**
+  Updates nodes around cells whose occupancy values have changed. Takes into
+  account the cspace of the robot. This step is performed after the robot
+  moves and the occupancy grid is updated with new sensor information
+
+  @return the number of nodes updated
+  */
+  int updateNodesAroundUpdatedCells();
+  /**
+  Constructs the optimal path through the search space by greedily moving to
+  the next position (or vertex) that minimizes the linearly interpolated
+  path cost calculation. Constructed paths populate the `path` attribute
+
+  @param[in] lookahead_dist number of future cell traversals to recurisively
+          analyze before moving to next position
+  */
+  void constructOptimalPath(int lookahead_dist);
+  /**
+  Helper method for path reconstruction process. Finds the next path position(s)
+  when planning from a vertex or an edge position on the graph given the current
+  position p and its consecutive neighbors p_a, p_b.
+
+  @param[in] p edge on graph to plan from
+  @param[in] p_a consecutive neighbor of p
+  @param[in] p_b consecutive neighbor of p
+  @return vector containing the next positions(s) and movement cost
+  */
+  path_additions computeOptimalCellTraversal(const Position& p, const Position& p_a, const Position& p_b);
+  /**
+  Helper method for path reconstruction process. Finds the next path position(s)
+  when planning from a vertex or an edge position on the graph.
+
+  @param[in] p edge on graph to plan from
+  @return vector containing the next positions(s) and movement cost
+  */
+  path_additions getPathAdditions(const Position& p, int lookahead_dist);
+  /**
+  Checks whether a specified node is within range of the goal node. This 'range'
+  is specified by the GOAL_RANGE instance variable.
+
+  @param[in] s Node to check
+  @return whether or not node s is within range of the goal
+  */
+  bool isWithinRangeOfGoal(const Position& p);
+  /**
+  Tries to insert an entry into the unordered map. If an entry for that key
+  (Node) already exists, overrides the value with specified g and rhs vals.
+
+  @param[in] s key to create entry for
+  @param[in] g g value for entry
+  @param[in] rhs rhs value for entry
+  */
+  void insert_or_assign(Node s, float g, float rhs);
```

Comment:
again, make up your mind snake_case or camelCase. I think you should convert everything to snake_case.

Reply:
I named this `insert_or_assign` because as of c++17, `std::unordered_map` has a member function called [`insert_or_assign`](https://en.cppreference.com/w/cpp/container/unordered_map/insert_or_assign). However, we still compile with c++11 and cannot make use of this feature so I created a method that implements this functionality with the same name. It does indeed look a bit out of place.

Reply:
Naming convention issues addressed in latest commit